src/Text/Pandoc/Writers/Typst.hs

```diff
@@ -88,6 +90,42 @@ pandocToTypst options (Pandoc meta blocks) = do
        Nothing  -> main
        Just tpl -> renderTemplate tpl context
 
+pickTypstAttrs :: [(Text, Text)] -> ([(Text, Text)],[(Text, Text)])
+pickTypstAttrs =
+  partitionEithers . catMaybes . map (\(k,v) ->
+    let spl = T.splitOn ":" k in
+    if length(spl) < 2 || length(spl) > 3
+      || head spl /= "typst"
+      || (length(spl) == 3 && spl !! 1 /= "text")
```

Comment:
We like to avoid using partial functions like `head`. I think it's cleaner to use pattern matching for this kind of thing. Could you do something like
```hs

case T.splitOn ":" k of
  "typst":"text":x -> Right (x,v)
  "typst":x -> Left (x, v)
  _ -> Nothing
```

Second thought: instead of using `partitionEithers . catMaybes . map`, why not just use a fold that directly constructs the tuple?  This way you also avoid needing to use Maybe or Either.

```hs
pickTypstAttrs = foldr go ([],[])
  where
    go (k,v) (eltProps, textProps) =
      case T.splitOn ":" k of
        "typst":"text":x -> (eltProps, (x,v):textProps)
        "typst":x -> ((x,v):eltProps, textProps)
        _ -> (eltProps, textProps)
```
Or even more compactly with Data.Bifunctor.(first,second)
```hs
pickTypstAttrs = foldr go ([],[])
  where
    go (k,v)
      case T.splitOn ":" k of
        "typst":"text":x -> second ((x,v):)
        "typst":x -> first ((x,v):)
        _ -> id
```

Reply:
This is very nice!

All five of my logical conditions covered by three patterns, and the meaning is very clear with the bifunctor approach.